### PR TITLE
fix(tools/v2fix): skip setting "pkg_path" on context if function call doesn't have any 

### LINKED ISSUE #3415: [BUG]: v2check panic
### Tracer Version(s)

gopkg.in/DataDog/dd-trace-go.v1 v1.72.2

### Go Version(s)

go version go1.24.2 darwin/arm64

### Bug Report

Hello,
I'm trying to use `v2check` to upgrade dd-trace-go in my codebase but it results in the following panic:

```
❯ v2check ./...
2025/04/11 09:44:48 Running v2check with 5 known changes
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x0 pc=0x10147a6f4]

goroutine 4704 [running]:
go/types.(*Package).Path(...)
        /opt/homebrew/Cellar/go/1.24.2/libexec/src/go/types/package.go:34
github.com/DataDog/dd-trace-go/tools/v2check/v2check.IsFuncCall({0x10191f3f8, 0x14010ad1260}, {0x101917c80?, 0x140040b1c40}, 0x14002d400e0?)
        /[REDACTED]/go/pkg/mod/github.com/!data!dog/dd-trace-go/tools/v2check@v0.0.2/v2check/probe.go:108 +0xe4
github.com/DataDog/dd-trace-go/tools/v2check/v2check.eval({0x101922c20, 0x1400026d200}, {0x101917c80, 0x140040b1c40}, 0x14002d400e0)
        /[REDACTED]/go/pkg/mod/github.com/!data!dog/dd-trace-go/tools/v2check@v0.0.2/v2check/known_change.go:99 +0xb0
github.com/DataDog/dd-trace-go/tools/v2check/v2check.Checker.runner.func1.1({0x101917c80, 0x140040b1c40})
        /[REDACTED]/go/pkg/mod/github.com/!data!dog/dd-trace-go/tools/v2check@v0.0.2/v2check/v2check.go:53 +0x8c
golang.org/x/tools/go/ast/inspector.(*Inspector).Preorder(0x140231b1a28, {0x1400d83fc20?, 0x1400d83fc28?, 0x100cdb460?}, 0x1400d83fc60)
        /[REDACTED]/go/pkg/mod/golang.org/x/tools@v0.29.0/go/ast/inspector/inspector.go:96 +0x98
github.com/DataDog/dd-trace-go/tools/v2check/v2check.Checker.runner.func1(0x14002d400e0)
        /[REDACTED]/go/pkg/mod/github.com/!data!dog/dd-trace-go/tools/v2check@v0.0.2/v2check/v2check.go:50 +0xf8
golang.org/x/tools/go/analysis/checker.(*Action).execOnce.func3(...)
        /[REDACTED]/go/pkg/mod/golang.org/x/tools@v0.29.0/go/analysis/checker/checker.go:355
golang.org/x/tools/go/analysis/checker.(*Action).execOnce(0x140035069a0)
        /[REDACTED]/go/pkg/mod/golang.org/x/tools@v0.29.0/go/analysis/checker/checker.go:376 +0x8f8
sync.(*Once).doSlow(0x1?, 0x0?)
        /opt/homebrew/Cellar/go/1.24.2/libexec/src/sync/once.go:78 +0xf0
sync.(*Once).Do(...)
        /opt/homebrew/Cellar/go/1.24.2/libexec/src/sync/once.go:69
golang.org/x/tools/go/analysis/checker.(*Action).exec(...)
        /[REDACTED]/go/pkg/mod/golang.org/x/tools@v0.29.0/go/analysis/checker/checker.go:263
golang.org/x/tools/go/analysis/checker.execAll.func1(0x0?)
        /[REDACTED]/go/pkg/mod/golang.org/x/tools@v0.29.0/go/analysis/checker/checker.go:251 +0x50
created by golang.org/x/tools/go/analysis/checker.execAll in goroutine 1
        /[REDACTED]/go/pkg/mod/golang.org/x/tools@v0.29.0/go/analysis/checker/checker.go:257 +0x14c
```

Thanks!

### Reproduction Code

_No response_

### Error Logs

_No response_

### Go Env Output

_No response_

### PR DESCRIPTION
**Pull request by darccio**
### What does this PR do?

Adds a nil check on `types.Func.Pkg`, as built-in functions and some edge cases don't have a package.

Also, refactors context keys to a proper type.

### Motivation

Fixes #3415 

### Reviewer's Checklist

- [ ] Changed code has unit tests for its functionality at or near 100% coverage.
- [ ] [System-Tests](https://github.com/DataDog/system-tests/) covering this feature have been added and enabled with the va.b.c-dev version tag.
- [ ] There is a benchmark for any new code, or changes to existing code.
- [ ] If this interacts with the agent in a new way, a system test has been added.
- [ ] New code is free of linting errors. You can check this by running `golangci-lint run` locally.
- [ ] Add an appropriate team label so this PR gets put in the right place for the release notes.
- [ ] Non-trivial go.mod changes, e.g. adding new modules, are reviewed by @DataDog/dd-trace-go-guild.

Unsure? Have a question? Request a review!

### CODE REVIEW

File: tools/v2fix/v2fix/probe.go [hunk 1]
```diff
@@ -103,9 +103,15 @@ func IsFuncCall(ctx context.Context, n ast.Node, pass *analysis.Pass) (context.C
...
 		return ctx, false
 	}
-	ctx = context.WithValue(ctx, "fn", fn)
-	ctx = context.WithValue(ctx, "args", c.Args)
-	ctx = context.WithValue(ctx, "pkg_path", fn.Pkg().Path())
+	ctx = context.WithValue(ctx, fnKey, fn)
+	ctx = context.WithValue(ctx, argsKey, c.Args)
+	pkg := fn.Pkg()
+	if pkg == nil {
+		// It could be a built-in function, in which case
```
Comment (kakkoyun) [inline tools/v2fix/v2fix/probe.go:110]: 👍

**Review by kakkoyun — COMMENTED** [review]
LGTM.

CI needs some love though.

**Review by kakkoyun — APPROVED** [review]
LGTM.

Thanks for splitting it into several commits, make it easier to review.
